### PR TITLE
Add vrouter agent customization

### PR DESCRIPTION
**Pull request by pierreblanc**
Just add new params to customize the /etc/contrail/contrail-vrouter-agent.conf file

### CODE REVIEW

**Comment by Spredzy** [comment]
:-1:

Thanks @pierreblanc for the review. However this PR is not suitable to use with this module.

When dealing with configuration file they are two ways to proceed - and they are mutually exclusive: 

  * Use puppet resource to specify a value within a file (the way this module is built)
  * Use a template configuration file (your PR)

Everything you're trying to do with this PR could already be done with the module.

Instead of passing `$max_vm_flows` as a parameter that will later be passed to the template you can use `contrail_vrouter_agent_config{"FLOWS/max_vm_flows": value => 42} `.

An helper parameters has been built for that[1]

So basically, within your hiera or main site.pp you should do something like : 

```
contrail::vrouter::config::vrouter_agent_config:
  FLOWS/max_vm_flows:
    value: 42
  FLOW/max_vm_linklocal_flows:
    value: 21
```

Note: my hiera exact structure might be slightly off, but this is the idea. Create a resource for each configuration and rely on `vrouter_agent_config` to gather them all.

Let me know if I haven't been clear enough, or anything else needs explanation
[1] https://github.com/redhat-cip/puppet-contrail/blob/master/manifests/vrouter/config.pp#L77

**Comment by pierreblanc** [comment]
Nice !

Merci Yanis. En effet c'est bien plus simple et flexible.
Un grand merci pour ces explications claires et précises!

Tchouss

----- Original Message -----
